### PR TITLE
job and shell script for cluster instance monitoring

### PR DESCRIPTION
**Pull request by hchen15**

### CODE REVIEW

File: devops/jobs/ClusterInstanceMonitoring.groovy [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+class ClusterInstanceMonitoring{
+	public static def job = { dslFactory, extraVars ->
+		assert extraVars.containsKey('DEPLOYMENTS') : "Please define DEPLOYMENTS. It should be a list of strings."
+        assert !(extraVars.get('DEPLOYMENTS') instanceof String) : "Make sure DEPLOYMENTS is a list and not a string"
+		extraVars.get('DEPLOYMENTS').each { deployment ->
+			
+			dslFactory.job(extraVars.get("FOLDER_NAME","Monitoring") + "/cluster-instance-monitoring-${deployment}") {
+				wrappers common_wrappers
+                logRotator common_logrotator
+
```
Comment (jibsheet) [inline devops/jobs/ClusterInstanceMonitoring.groovy]: Some super wacky formatting on lines 16-21, probably mixed tabs and spaces?

File: devops/jobs/ClusterInstanceMonitoring.groovy [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+                wrappers {
+                    credentialsBinding {
+                        file('AWS_CONFIG_FILE','tools-edx-jenkins-aws-credentials')
+                        def role = "find-active-instances-${deployment}-role-arn"
```
Comment (jibsheet) [inline devops/jobs/ClusterInstanceMonitoring.groovy]: I know we came up with this workaround when we were debugging '' vs "".  Do you still need it?

File: devops/jobs/ClusterInstanceMonitoring.groovy [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+                        string('ROLE_ARN', role)
+                    }
+                }
+
+                def gitCredentialId = extraVars.get('SECURE_GIT_CREDENTIALS','')
+
+                def config_internal_repo = "git@github.com:edx/${deployment}-internal.git"
+                if (deployment == 'mckinsey'){
```
Comment (jibsheet) [inline devops/jobs/ClusterInstanceMonitoring.groovy]: We should avoid hardcoding these folks in a public repo.  I'd vote either making this private or figuring out another way to do it from the config block

File: devops/jobs/ClusterInstanceMonitoring.groovy [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+                            cleanAfterCheckout()
+                            pruneBranches()
+                            relativeTargetDirectory('configuration-internal')
+                        }
+                    }
+                }
+
+                throttleConcurrentBuilds {
+                    maxPerNode(0)
+                    maxTotal(0)
```
Comment (jibsheet) [inline devops/jobs/ClusterInstanceMonitoring.groovy]: Does this turn off building or disable limits by setting both to 0

File: devops/jobs/ClusterInstanceMonitoring.groovy [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+                        command(
+                            dslFactory.readFileFromWorkspace("devops/resources/cluster-instance-monitoring.sh")
+                        )
+
+                    }
+
+                }
+
+                publishers {
+                    mailer(extraVars.get('NOTIFY_ON_FAILURE','devops@edx.org'), false, false)
```
Comment (jibsheet) [inline devops/jobs/ClusterInstanceMonitoring.groovy]: Don't hardcode us into a public repo, better to conditionally set publishers based on NOTIFY_ON_FAILURE being empty or not.

File: devops/jobs/ClusterInstanceMonitoring.groovy (hunk 4 above)
Comment (hchen15) [inline devops/jobs/ClusterInstanceMonitoring.groovy]: it disables the limits

File: devops/jobs/ClusterInstanceMonitoring.groovy [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+                if (configuration.get('internal_repo')){
+                    internal_config = configuration.get('internal_repo')
+                }
+
+                extraVars['CONFIGURATION_INTERNAL_REPO'] = internal_config
+
+                parameters{
+                    stringParam('CONFIGURATION_REPO', extraVars.get('CONFIGURATION_REPO', 'https://github.com/edx/configuration.git'),
+                                    'Git repo containing edX configuration.')
+                    stringParam('CONFIGURATION_BRANCH', extraVars.get('CONFIGURATION_BRANCH', 'hchen/cluster-monitoring'),
```
Comment (feanil) [inline devops/jobs/ClusterInstanceMonitoring.groovy]: This will need to be set to master before this merges.

**Pull request merged**